plugins/quotes.py

```diff
@@ -0,0 +1,217 @@
+#quotes
+#
+#usage:
+# .qinit                                       --- database initialization command (admin only)
+# .q [nick/channel] [optional quote number]    --- for fetching quotes
+# .q add [nick] [quote]                        --- for adding quotes
+# .q delete [quote number]                     --- for deleting quotes (you can only delete your own quotes)
+#
+#made by kelp
+
+from typing import Optional, Dict, Any, List, Set, Union, Callable, Tuple
+from dataclasses import dataclass, field
+import time
+import asyncio
+import random
+import requests
+
+#import core db functions
+from core import db
+
+#db table definition
+quote_columns: List[str] = ['chan', 'nick', 'add_nick', 'msg', 'time', 'deleted']
+
+def remove_quotes(quotelist):
+	"""removes quotes from a quotelist where deleted == '1' """
+	buf = []
+	for quote in quotelist:
+		if quote[5] != '1':
+			buf.append(quote)
+			
+	return buf
+	
+def display_quote(client, data, quotelist, target, arg):
+	"""
+	output a quote, given a tuple list of all avalable quotes, and possibly
+	a number (for selection of a specific quote)
+	"""
+	
+	outquotes = []
+	quotenums = []
+	
+	if arg is None:
+		"""pick a random number from the quotelist if there is no argument provided"""
+		print("random numarg picked.")
+		numarg = random.randint(0, len(quotelist)-1)
+	else:
+		try:
+			if int(arg) > 0:
+				numarg = int(arg)-1
+			elif int(arg) < 0:
+				numarg = int(arg)
+			else:
+				numarg = 0
+		except:
+			wordarg = arg
+	
+	
+	quotelist = remove_quotes(quotelist)
+	
+	if len(quotelist) < 1:
+		asyncio.create_task(client.message(data.target, 'I have no quotes for '+target))
+		return
+		
+	if 'numarg' in locals():
```

Comment:
Could just set numarg and wordarg to None, then do `if not thing` since it should be less overhead than looking into locals()